design/pipeline-artifacts.md

```diff
@@ -33,20 +33,19 @@ Artifacts are distinct from other `resources` types, including `containers`, `re
 
 ---
 
-## Downloading artifacts: `downloadArtifact`
+## Downloading artifacts: `download`
 
-`downloadArtifact` is a shortcut for several tasks:
+`download` is a shortcut for two tasks:
 
 - The [Download Pipeline Artifacts](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/download-pipeline-artifact) task
 - The [Download Fileshare Artifacts](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/download-fileshare-artifacts) task
-- Download tasks for other pipelines systems e.g. Jenkins
 
-It will download artifacts published from a previous job or stage or from another pipeline. Artifacts are downloaded either to `$PIPELINES_RESOURCESDIR` or to the directory specified in `root`.
+It will download artifacts uploaded from a previous job or stage or from another pipeline. Artifacts are downloaded either to `$PIPELINES_RESOURCESDIR` or to the directory specified in `root`.
 
 ### Schema
 
 ```yaml
-- downloadArtifact: string # identifier for the resource from which to download artifacts, optional; defaults to 'self`
+- download: string # identifier for the pipeline resource from which to download artifacts, optional; defaults to `self`, which represents the current pipeline
```

Comment:
Is it weird to use `self` here? In a yaml build, `self` is the repo where the entry yaml file is defined. Would it be better to say `""` means the current pipeline rather than overload the concept `self`?

Reply:
Or perhaps make it explicit but not use the keyword `self` -- how bad would it be to squat on `current`?

Reply:
I wonder if we need to make resource aliases distinct across categories? If so, then this should not be `self`.

Also note, the name will matter when we expose resources into the expression context. Currently only resources are exposed and only for the extension injection stuff.

Reply:
I like leaving it blank - changed.

Reply:
Has consistency been considered? In other words, should we allow `checkout: self` to be consistent?

Personally, I haven't heard users complain about `checkout: self`. And generally I think we confuse users when we have multiple ways of doing the same thing.

Reply:
+1 to @ericsciple's point, I don't like blank implying a behavior.